app/controllers/votes_controller.rb

```diff
@@ -8,9 +8,13 @@ def upvote
       @value.vote = 1
     end
     respond_to do |format|
-      @value.save
-      @vote_total = Vote.group(:review_id).sum(:vote)
-      format.json { render json: @vote_total[@review.id] }
+      if @value.save
+        @vote_total = Vote.group(:review_id).sum(:vote)
+        format.json { render json: @vote_total[@review.id] }
+      else
+        flash[:error] = "You must be signed in"
+      end
+      format.html { redirect_to bookstore_path(@review.bookstore) }
```

Comment:
Ideally there will be an API endpoint in the app that serves the JS and a controller in here that serves html. Mixing the two is messy. Check out [an example](https://github.com/mlg-/surrealist_poetry_api).